System/Posix/Process/Common.hsc

```diff
@@ -383,9 +383,14 @@ readWaitStatus wstatp = do
 
 -- | @'exitImmediately' status@ calls @_exit@ to terminate the process
 --   with the indicated exit @status@.
---   The operation never returns.
-exitImmediately :: ExitCode -> IO ()
-exitImmediately exitcode = c_exit (exitcode2Int exitcode)
+--   The operation never returns. Since it does not use the Haskell exception
+--   system and it cannot be caught.
+exitImmediately :: ExitCode -> IO a
+exitImmediately exitcode = do
+	_ <- c_exit (exitcode2Int exitcode)
+	-- The above will exit the program, but need the following to satisfy
+	-- the type signature.
+	exitFailure
```

Comment:
I think you can recursively call `exitImmediately status` here. This code should be discovered as dead, so everything of the right type will do.

Reply:
That is a nice idea